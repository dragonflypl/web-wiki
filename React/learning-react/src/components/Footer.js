import React from 'react';
import FilterLink from './FilterLink';

export default function Footer() {
  return (<p>
    Show:&nbsp;
    <FilterLink filter='SHOW_ALL'>All</FilterLink>,&nbsp;
    <FilterLink filter='SHOW_COMPLETED'>Completed</FilterLink>,&nbsp;
    <FilterLink filter='SHOW_ACTIVE'>Active</FilterLink>
  </p>)
}
