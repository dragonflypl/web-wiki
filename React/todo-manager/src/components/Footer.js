import React from 'react';
import FilterLink from './FilterLink';
import RandomNumber from '../services/RandomNumber';

const randomInRange = new RandomNumber(10, 20);

export default function Footer() {
  return <p>
    Show: {' '}
    <FilterLink filter='all'>All</FilterLink>{' '}
    <FilterLink filter='completed'>Completed</FilterLink>{' '}
    <FilterLink filter='active'>Active</FilterLink>
    <h2>This is random number: {randomInRange.generate()}</h2>
  </p>
}
