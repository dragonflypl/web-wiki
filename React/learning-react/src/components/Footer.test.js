jest.mock('../services/RandomNumber');

// React, mount, shallow, render is available on globals thanks to setupTests.js

import Footer from './Footer';
import TestRenderer from 'react-test-renderer';
import ShallowRenderer from 'react-test-renderer/shallow';

describe('Footer', () => {

  it('should render with shallow', () => {
    const wrapper = shallow(<Footer />)
    expect(wrapper).toMatchSnapshot();
  });

  it.skip('should render with mount', () => {
    const wrapper = mount(<Footer />)
    expect(wrapper).toMatchSnapshot();
  });

  it.skip('should render with render', () => {
    const wrapper = render(<Footer />)
    expect(wrapper).toMatchSnapshot();
  });

  it.skip('should render with TestRenderer', () => {
    const component = TestRenderer.create(<Footer />) ;
    expect(component.toJSON()).toMatchSnapshot();
  });

  it('should render with ShallowRenderer', () => {
    const renderer = new ShallowRenderer();
    renderer.render(<Footer />) ;
    expect(renderer.getRenderOutput()).toMatchSnapshot();
  });

})
