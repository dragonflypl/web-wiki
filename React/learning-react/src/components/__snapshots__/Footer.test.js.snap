// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render with ShallowRenderer 1`] = `
<p>
  Show: 
  <Connect(Link)
    filter="SHOW_ALL"
  >
    All
  </Connect(Link)>
  , 
  <Connect(Link)
    filter="SHOW_COMPLETED"
  >
    Completed
  </Connect(Link)>
  , 
  <Connect(Link)
    filter="SHOW_ACTIVE"
  >
    Active
  </Connect(Link)>
  <h2>
    This is random number: 
    10
  </h2>
</p>
`;

exports[`should render with shallow 1`] = `
<p>
  Show: 
  <Connect(Link)
    filter="SHOW_ALL"
  >
    All
  </Connect(Link)>
  , 
  <Connect(Link)
    filter="SHOW_COMPLETED"
  >
    Completed
  </Connect(Link)>
  , 
  <Connect(Link)
    filter="SHOW_ACTIVE"
  >
    Active
  </Connect(Link)>
  <h2>
    This is random number: 
    10
  </h2>
</p>
`;
